test/Unosquare.Swan.Test/CsvWriterTest.cs

```diff
@@ -137,35 +172,125 @@ public void Strings_ReturnsAreEqual()
+                    Assert.AreEqual(1, (int)writer.Count);
+                }
+            }
+        }
     }
 
     [TestFixture]
     public class WriteHeadings : CsvWriterTest
     {
         [Test]
-        public void HeadersLength_ReturnsAreEqual()
+        public void NullType_ThrowsArgumentNullException()
+        {            
+            using (var stream = new MemoryStream())
+            {
+                using (var writer = new CsvWriter(stream))
+                {
+                    Assert.Throws<ArgumentNullException>(() =>
+                    {
+                        writer.WriteHeadings(null as Type);
+                    });
+                }
+            }
+        }
+
+        [Test]
+        public void NullDictionary_ThrowsArgumentNullException()
         {
-            var tempFile = Path.GetTempFileName();
-            var headers = new[]
+            using (var stream = new MemoryStream())
             {
-                "AccessDate", "AlternateId", "CreationDate", "Description", "Id", "IsValidated", "Name", "Score",
-                "ValidationResult"
+                using (var writer = new CsvWriter(stream))
+                {
+                    Assert.Throws<ArgumentNullException>(() =>
+                    {
+                        writer.WriteHeadings(null as Dictionary<string, string>);
+                    });
+                }
+            }
+        }
+
+        [Test]
+        public void HeadersDictionaryLength_ReturnsAreEqual()
+        {
+            var dictionaryHeaders = new Dictionary<string, string>
+            {
+                {"AccessDate", "20171107"},
+                {"AlternateId", "1"},
+                {"CreationDate", "20171107"},
+                {"Description", "Sr. Software Engineer"},
+                {"Id", "0001"},
+                {"IsValidated", "true"},
+                {"Name", "Alexey Turpalov"},
+                {"Score", "1245F"},
+                {"ValidationResult", "true"}
             };
+            var stringHeadersOutput = "AccessDate,AlternateId,CreationDate,Description,Id,IsValidated,Name,Score,ValidationResult\r\n";
 
-            using (var stream = File.OpenWrite(tempFile))
+            var stringHeaders = dictionaryHeaders.Select(k => k.Key).ToList();
+
+            using (var stream = new MemoryStream())
+            {
+                using (var writer = new CsvWriter(stream))
+                {
+                    writer.WriteHeadings(dictionaryHeaders);
+
+                    stream.Position = 0;
+                    var sr = new StreamReader(stream);
+                    var myStr = sr.ReadToEnd();
+                    var myStrSplitted = myStr.Split(',');
+
+                    Assert.AreEqual(stringHeadersOutput, myStr);
+                    Assert.AreEqual(stringHeaders.Count, myStrSplitted.Length);
+                }
+            }
+        }
+
+        [Test]
+        public void HeadersLength_ReturnsAreEqual()
```

Comment:
Not sure about the name of the test. Please look at the unit testing best practices wiki.